File: DSA/SuryanshMishra_590012069/Day72/day72question1.cpp
#include <cstddef>
using namespace std;
struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};
TreeNode* mirrorTree(TreeNode* root) {
    if (root == nullptr) return nullptr;
    TreeNode* tmp = root->left;
    root->left = root->right;
    root->right = tmp;
    mirrorTree(root->left);
    mirrorTree(root->right);

    return root;
}

File: DSA/akashyadav_590014126/Day_72_Question_1.py
class TreeNode:
    def __init__(self,val=0,left=None,right=None):
        self.val=val
        self.left=left
        self.right=right

def mirror_tree(root):
    if root is None:
        return None
    
    root.left,root.right=root.right,root.left

    mirror_tree(root.left)
    mirror_tree(root.right)

    return root